File: webSISGERT/ClientApp/src/app/ordenes-trabajo/ordenTrabajo.ts
import { ICliente } from "../clientes/cliente";
import { iDetalleTarea } from "./detalle-tarea/detalleTarea";
import { IDetallePago } from "./detalle-pago/detallePago";
import { IDetalleEmpleado } from "./detalle-empleado/detalleEmpleado";
import { iMaquinaria } from "../maquinarias/maquinaria";
import { iDetalleCosto } from "./detalle-costo/detalleCosto";
import { IDetalleRepuesto } from "./detalle-repuesto/detalleRepuesto";
import { ITipoOT } from "../tipos-ot/tipoOT";

export enum EstadoOT {
  Activo,
  Cancelado,
  Cerrado
}

export interface iOrdenTrabajo {
  id: number,
  tipoOTId:string,
  tipoOT: ITipoOT,
  codigo: string,
  clienteId: number,
  cliente: ICliente,
  favorito: boolean,
  maquinariaId: number,
  maquinaria: iMaquinaria,
  fecha: Date,
  descripcion: string,
  observaciones: string,
  informePreliminar: string,
  formatoRecepcionEquipos: string,
  actaConformidad: string,
  estadoOT: EstadoOT,
  detallesTareas: iDetalleTarea[],
  detallesPagos: IDetallePago[],
  detallesEmpleados: IDetalleEmpleado[],
  detallesCosto: iDetalleCosto[],
  detallesRepuestos: IDetalleRepuesto[],
  sumaPagos: number,
  sumaTareas: number,
  sumaCostos: number,
  sumaRepuesto:number,
  TotalCosto:number,
}

File: webSISGERT/Models/OT/OrdenTrabajo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;

namespace webSISGERT.Models.OT
{
    public class OrdenTrabajo
    {
        public enum EstadoOT
        {
            Activo,
            Cancelado,
            Cerrado
        }
        public int Id { get; set; }
        [StringLength(maximumLength: 15, ErrorMessage = "El numero maximo de caracteres para el codigo es 15.")]
        public string Codigo { get; set; }
        public Boolean Favorito { get; set; }
        public int ClienteId { get; set; }
        public virtual Cliente cliente { get; set; }
        public int MaquinariaId { get; set; }
        public virtual Maquinaria maquinaria {get;set;}
        public DateTime Fecha { get; set; }
        public string Descripcion { get; set; }
        public string Observaciones { get; set; }
        public string InformePreliminar { get; set; }
        public string FormatoRecepcionEquipos { get; set; }
        public string ActaConformidad { get; set; }
        public EstadoOT Estado { get; set; }
        public virtual ICollection<DetalleTarea> DetallesTareas { get; set; }
        public virtual ICollection<DetalleEmpleado> DetallesEmpleados { get; set; }
        public virtual ICollection<DetalleRepuesto> DetallesRepuestos { get; set; }
        public virtual ICollection<DetallePago> DetallesPagos { get; set; }
        public virtual ICollection<DetalleCosto> DetallesCosto { get; set; }
    }
}
